src/coprocessor/dag/rpn_expr/impl_time.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use cop_codegen::rpn_fn;
+
+use super::super::expr::EvalContext;
+
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::Error;
+use crate::coprocessor::Result;
+
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn date_format(
+    ctx: &mut EvalContext,
+    t: &Option<DateTime>,
+    layout: &Option<Bytes>,
+) -> Result<Option<Bytes>> {
+    use std::str::from_utf8;
+
+    if t.is_none() || layout.is_none() {
+        return Ok(None);
+    }
+    let (t, layout) = (t.as_ref().unwrap(), layout.as_ref().unwrap());
+    if t.invalid_zero() {
```

Comment:
Any test case cover it?

Reply:
Tests about None are added in latest commits.